scenarios/scenario_build.rs

```diff
@@ -1,9 +1,24 @@
-use happ_builder::{build_happs, required_tools_available, BuildOptions, HappBuilderResult};
+use happ_builder::{
+    anyhow, build_happs, required_tools_available, BuildOptions, HappBuilderResult,
+};
+use std::{path::PathBuf, str::FromStr};
 
 /// Build the required DNA(s) and hApps for the scenario.
 /// The built DNA(s) will appear as `/dnas/<scenario-name>/<dna-name>.dna` from the project root.
-/// /// The built hApp(s) will appear as `/happs/<scenario-name>/<happ-name>.happ` from the project root.
+/// The built hApp(s) will appear as `/happs/<scenario-name>/<happ-name>.happ` from the project root.
 fn main() -> HappBuilderResult {
+    let scenario_dir = PathBuf::from_str(std::env!("CARGO_MANIFEST_DIR")).unwrap();
+    let dir_name = scenario_dir.file_name().unwrap().to_str().unwrap();
+    let package_name = std::env!("CARGO_PKG_NAME");
+    if package_name != dir_name {
+        // The directory name is used as the nix package name which is in turn used with `cargo run`
+        // inside the nix package. Therefore, fail the build if the directory name does not match the
+        // package name.
+        anyhow::bail!(
```

Comment:
Isn't a panic equally informative here? It's in a `main` so it's always at the top level with no context to track and it's a build script rather than a 'user-facing' command line tool. I'd prefer to avoid the `anyhow` dependency here

Reply:
Sure, I was torn because this way adds a dependency but it has a cleaner stderr:
```console
 Error: The package name of the scenario 'app_install' does not match the directory name 'app_installer'
```
vs a panic:
```console
thread 'main' panicked at scenarios/app_installer/../scenario_build.rs:17:9:
The package name of the scenario 'app_install' does not match the directory name 'app_installer'
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```